
project build/
diff --git a/qemu.mk b/qemu.mk
index 038e982..0c22250 100644
--- a/qemu.mk
+++ b/qemu.mk
@@ -166,14 +166,9 @@ QEMU_SMP ?= 2
 .PHONY: run-only
 run-only:
 	ln -sf $(ROOT)/out-br/images/rootfs.cpio.gz $(BINARIES_PATH)/
-	$(call check-terminal)
-	$(call run-help)
-	$(call launch-terminal,54320,"Normal World")
-	$(call launch-terminal,54321,"Secure World")
-	$(call wait-for-ports,54320,54321)
 	cd $(BINARIES_PATH) && $(QEMU_PATH)/arm-softmmu/qemu-system-arm \
 		-nographic \
-		-serial tcp:localhost:54320 -serial tcp:localhost:54321 \
+		-serial telnet:localhost:50001,server,nowait -serial telnet:localhost:50002,server,nowait \
 		-smp $(QEMU_SMP) \
 		-s -S -machine virt,secure=on -cpu cortex-a15 \
 		-d unimp -semihosting-config enable,target=native \
