
project build/
diff --git a/build/qemu.mk b/build/qemu.mk
index 038e982..b5cf100 100644
--- a/build/qemu.mk
+++ b/build/qemu.mk
@@ -166,16 +166,11 @@ QEMU_SMP ?= 2
 .PHONY: run-only
 run-only:
 	ln -sf $(ROOT)/out-br/images/rootfs.cpio.gz $(BINARIES_PATH)/
-	$(call check-terminal)
-	$(call run-help)
-	$(call launch-terminal,54320,"Normal World")
-	$(call launch-terminal,54321,"Secure World")
-	$(call wait-for-ports,54320,54321)
 	cd $(BINARIES_PATH) && $(QEMU_PATH)/arm-softmmu/qemu-system-arm \
 		-nographic \
-		-serial tcp:localhost:54320 -serial tcp:localhost:54321 \
+		-serial telnet::50001,server -serial telnet::50002 \
 		-smp $(QEMU_SMP) \
-		-s -S -machine virt,secure=on -cpu cortex-a15 \
+		-s -machine virt,secure=on -cpu cortex-a15 \
 		-d unimp -semihosting-config enable,target=native \
 		-m 1057 \
 		-bios bl1.bin \
